libs/GraphicECS/SFML/Resources/SoundResource.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+** EPITECH PROJECT, 2022
+** R-Type
+** File description:
+** SoundResources
+*/
+
+#ifndef SOUNDRESOURCES_HPP_
+#define SOUNDRESOURCES_HPP_
+
+#include <SFML/Audio.hpp>
+#include <filesystem>
+#include "Resource/Resource.hpp"
+#include <unordered_map>
+#include "Error/Error.hpp"
+
+namespace graphicECS::SFML::Resources
+{
+    /// @brief This resource class stores a map of Sounds SFML.
+    class SoundResource : public ecs::Resource {
+      public:
+        /// @brief Enumeration of all available Sounds
+        enum soundName_e { UNDEFINED, SHOOT, DEATH, EXPLOSION };
+
+        /// @brief Name of map which contains Sounds.
+        using SoundsList = std::unordered_map<soundName_e, std::shared_ptr<sf::Sound>>;
+
+        /// @brief Add a Sound from it's Sound Path passed as parameter
+        /// @param sound_e Enum of the Sound
+        /// @param soundPath The sound path to be used.
+        inline SoundResource(const soundName_e sound_e, const std::filesystem::path &soundPath)
```

Comment:
May add a possibility to create a resource without sound and add them later (example : if you have some enemies which aren't loaded/used) ?

Reply:
Nop all song and sounds must be loaded at the creation of the world in the music/sound Resource

Reply:
Why must they be created at initialization?

Reply:
Music are loaded at start like texture, fonts...